### PR TITLE
Remove everything about chunking

### PR DESCRIPTION
**Pull request by zasdfgbnm**
It is not only WIP but actually just started
- [x] Remove chunking in doc string in `torchani/aev.py`
- [x] Remove chunking in old dataset API (code)
- [x] Remove chunking in old dataset API (doc string)
- [x] Remove chunking in `tests/test_data.py`
- [x] Remove chunking in new dataset API (CachedDataset)
- [x] Remove chunking in new dataset API (CachedDataset doc string)
- [x] Remove chunking in new dataset API (ShuffledDataset)
- [x] Remove chunking in new dataset API (ShuffledDataset doc string)
- [x] Remove chunking in `tests/test_data_new.py`
- [x] Merge old and new dataset API, there will be no longer old/new difference.
- [x] Merge `tests/test_data.py` and `tests/test_data_new.py`
- [x] Remove chunking in `tests/test_jit_builtin_models.py`
- [x] Remove chunking in `examples/nnp_training.py`
- [x] Remove chunking in `examples/nnp_training_force.py`
- [x] Remove chunking in NeuroChem trainer
- [x] Remove chunking in `tools/training-benchmark-nsys-profile.py`
- [x] Remove chunking in `tools/training-benchmark.py`

### CODE REVIEW

**Comment by IgnacioJPickering** [comment]
@zasdfgbnm wait! chunking is required right?

**Comment by IgnacioJPickering** [comment]
Otherwise the dataset will take up a ton of space in memory right?

**Comment by IgnacioJPickering** [comment]
I thought that was the  whole idea of why chunking was implemented

**Comment by zasdfgbnm** [comment]
@IgnacioJPickering Actually not, the tons of space is because the old dataset API is implemented stupidly (it pad to the largest number of atoms in the dataset).

**Comment by IgnacioJPickering** [comment]
So you mean we can pad to largest atom in the batch and we are good without chunking?

**Comment by IgnacioJPickering** [comment]
That would be great

File: examples/nnp_training.py [hunk 1]
```diff
@@ -80,37 +80,17 @@
 except NameError:
     path = os.getcwd()
 dspath = os.path.join(path, '../dataset/ani1-up_to_gdb4/ani_gdb_s01.h5')
-
 batch_size = 2560
 
-training, validation = torchani.data.load_ani_dataset(
-    dspath, species_to_tensor, batch_size, rm_outlier=True, device=device,
-    transform=[energy_shifter.subtract_from_dataset], split=[0.8, None])
+dataset = torchani.data.load(dspath).subtract_self_energies(energy_shifter).remove_outliers().species_to_indices().shuffle()
```
Comment (farhadrgh) [inline examples/nnp_training.py]: I think we can take out the default `remove_outliers()` from example files. The ANI1x and ANI2x datasets now have the outlier energy conformers removed. And in the case of custom datasets, the user may come up with different thresholds than defaults.

Comment (zasdfgbnm) [reply examples/nnp_training.py]: fixed

**Comment by zasdfgbnm** [comment]
**Computer:** AMD Threadripper 3970x + NVIDIA 2080Ti
**Timing:** 1 epoch = ~43s
**Convergence:** 62 epoches = RMSE 3.4 kcal/mol

Looks good to me, merging now.

**Pull request merged**